src/Compilers/Core/Portable/Text/ChangedText.cs

```diff
@@ -240,149 +240,171 @@ private static ImmutableArray<TextChangeRange> Merge(IReadOnlyList<ImmutableArra
+                }
+                else if (newChange.Span.End < oldChange.Span.Start + oldDelta)
+                {
+                    // new change is entirely before old change
+                    addAdjustedNewChange(newChange);
+                    if (tryGetNext(newChanges, ref newIndex, out newChange)) continue; else break;
+                }
+                else if (newChange.Span.Start > oldChange.Span.Start + oldDelta + oldChange.NewLength)
+                {
+                    // new change is entirely after old change
+                    addAndAdjustOldDelta(oldChange);
+                    if (tryGetNext(oldChanges, ref oldIndex, out oldChange)) continue; else break;
+                }
+                else if (newChange.Span.Start < oldChange.Span.Start + oldDelta)
+                {
+                    // new change starts before old change, but overlaps
+                    // add as much of new change deletion as possible and try again
+                    var newChangeLeadingDeletion = oldChange.Span.Start + oldDelta - newChange.Span.Start;
+                    add(new TextChangeRange(new TextSpan(newChange.Span.Start - oldDelta, newChangeLeadingDeletion), 0));
+                    newChange = new TextChangeRange(new TextSpan(oldChange.Span.Start + oldDelta, newChange.Span.Length - newChangeLeadingDeletion), newChange.NewLength);
+                    continue;
+                }
+                else if (newChange.Span.Start > oldChange.Span.Start + oldDelta)
+                {
+                    // new change starts after old change, but overlaps
+                    // add as much of the old change as possible and try again
+                    var oldChangeLeadingInsertion = newChange.Span.Start - (oldChange.Span.Start + oldDelta);
+                    var oldChangeLeadingDeletion = Math.Min(oldChange.Span.Length, oldChangeLeadingInsertion);
+                    addAndAdjustOldDelta(new TextChangeRange(new TextSpan(oldChange.Span.Start, oldChangeLeadingDeletion), oldChangeLeadingInsertion));
+                    oldChange = new TextChangeRange(new TextSpan(newChange.Span.Start - oldDelta, oldChange.Span.Length - oldChangeLeadingDeletion), oldChange.NewLength - oldChangeLeadingInsertion);
+                    continue;
+                }
+                else
+                {
+                    // new change and old change start at same position
+                    if (oldChange.NewLength == 0)
                     {
-                        // new change is a non-change, just ignore it and move on
-                        newIndex++;
-                        goto nextNewChange;
+                        // old change is just a deletion, go ahead and old change now and deal with new change separately
+                        addAndAdjustOldDelta(oldChange);
+                        if (tryGetNext(oldChanges, ref oldIndex, out oldChange)) continue; else break;
```

Comment:
>continue; else break [](start = 81, length = 20)

These two code paths appear to be hit by `TextChangeTests.Fuzz()` only. Should we have specific tests that also test these cases? Same for the `continue` in the final `else`.

Reply:
Have added the coverage, thanks!